C# backend: AsUnmanaged() is called on slice classes (e.g. SliceByte) that only define ToUnmanaged()

## Summary

The C# backend (`interoptopus_csharp` 0.16.0) generates `AsUnmanaged()` method bodies on composite types (structs / tagged enums) that call `.AsUnmanaged()` on embedded slice fields, but it never generates an `AsUnmanaged()` method on the slice classes themselves. A generated slice class such as `SliceByte` only gets `ToUnmanaged()`, so the produced C# fails to compile with `CS1061`.

Minimal public reproduction: https://github.com/sandersaares/whatiswrongwithyourdog-interoptopus-asunmanaged-tounmanaged

## Reproduction

A composite `#[ffi]` type that (1) embeds a slice (e.g. `ffi::Slice<u8>`) **and** (2) is itself used as the element type of a slice parameter triggers the bug:

```rust
#[ffi]
#[derive(Clone, Copy)]
pub enum AttributeValue<'a> {
    Bool(bool) = 0,
    I64(i64) = 1,
    String(ffi::Slice<'a, u8>) = 2,
}

#[ffi]
#[derive(Clone, Copy)]
pub struct Attribute<'a> {
    pub key: ffi::Slice<'a, u8>,
    pub tagged_value: AttributeValue<'a>,
}

// Using `Attribute` as a slice element forces the backend to marshal it via `AsUnmanaged()`.
#[ffi]
pub fn record_attributes(attributes: ffi::Slice<Attribute>) -> u32 {
    attributes.len() as u32
}
```

Generating and compiling the bindings (`cargo run --bin generate && dotnet build csharp`) fails:

```
error CS1061: 'SliceByte' does not contain a definition for 'AsUnmanaged' and no accessible extension method 'AsUnmanaged' accepting a first argument of type 'SliceByte' could be found
```

## The asymmetry

`AsUnmanaged()` is emitted for composite element types but not for slice element types:

| Generated type | `ToUnmanaged()` | `AsUnmanaged()` |
| --- | :---: | :---: |
| `AttributeValue` (tagged enum embedding a slice) | yes | yes |
| `Attribute` (struct embedding a slice) | yes | yes |
| `SliceByte` (slice class) | yes | **no** |

`Attribute.AsUnmanaged()` calls `key.AsUnmanaged()` and `AttributeValue.AsUnmanaged()` calls `_String.AsUnmanaged()`, where both `key` and `_String` are `SliceByte` — so the generated code references a method that was never emitted.

## Root cause

The C# backend has two slice templates. `templates/rust/pattern/slice/marshalling.cs` (used for non-blittable element types) already defines `internal Unmanaged AsUnmanaged() => ToUnmanaged();`, but `templates/rust/pattern/slice/fast.cs` (used for blittable element types — `SliceByte`, `SliceU32`, `SliceVec3f32`, …) defines only `ToUnmanaged()`. Composites unconditionally call `AsUnmanaged()` on every slice field, so any composite that embeds a blittable slice produces code that does not compile.

This path appears to be untested: in `reference_project` no composite embeds a blittable slice, so the dotnet compile gate never exercised `fast.cs` slices inside a composite.

## Expected vs. actual

- **Expected:** the generated C# compiles. The slice class should expose `AsUnmanaged()`, forwarding to `ToUnmanaged()` for a borrowed slice (neither method transfers ownership of the backing buffer) — matching what `marshalling.cs` already does.
- **Actual:** the backend emits `<compositeType>.AsUnmanaged()` bodies that call `<sliceField>.AsUnmanaged()`, but never emits `AsUnmanaged()` on slice classes generated from `fast.cs`.

## Fix

I have a fix that mirrors `marshalling.cs` in `fast.cs` and adds `reference_project` regression coverage (a struct embedding `Slice<u8>`, exercising the previously-uncovered path through the dotnet compile gate). PR to follow.

--- a/crates/backend_csharp/templates/rust/pattern/slice/fast.cs
+++ b/crates/backend_csharp/templates/rust/pattern/slice/fast.cs
@@ -98,6 +98,10 @@ public partial class {{ name }} : IEnumerable<{{ element_type }}>, IDisposable
         return unmanaged;
     }
 
+    {{ _fns_decorators_all | indent }}
+    {{ _fns_decorators_internal | indent }}
+    internal Unmanaged AsUnmanaged() => ToUnmanaged();
+
     [CustomMarshaller(typeof({{ name }}), MarshalMode.Default, typeof(Marshaller))]
     private struct MarshallerMeta { }
 

--- a/crates/reference_project/src/lib.rs
+++ b/crates/reference_project/src/lib.rs
@@ -154,6 +154,7 @@ pub fn inventory() -> RustInventory {
         // .register(function!(patterns::slice::pattern_ffi_slice_7))
         .register(function!(patterns::slice::pattern_ffi_slice_8))
         .register(function!(patterns::slice::pattern_ffi_slice_9))
+        .register(function!(patterns::slice::pattern_ffi_slice_in_struct))
         .register(function!(patterns::slice::pattern_ffi_slice_delegate))
         .register(function!(patterns::slice::pattern_ffi_slice_delegate_huge))
         .register(function!(patterns::option::pattern_ffi_option_1))

--- a/crates/reference_project/src/patterns/slice.rs
+++ b/crates/reference_project/src/patterns/slice.rs
@@ -87,3 +87,16 @@ pub fn pattern_ffi_slice_8(slice: &SliceMut<CharArray>, callback: CallbackCharAr
 pub fn pattern_ffi_slice_9(slice: Slice<UseString>) -> ffi::String {
     slice.as_slice()[0].s1.clone()
 }
+
+/// A composite embedding a blittable `Slice<u8>`. Its generated `AsUnmanaged()`
+/// calls `bytes.AsUnmanaged()`, so the `SliceByte` binding must expose that method.
+#[ffi]
+#[derive(Clone)]
+pub struct UseSliceByteInStruct<'a> {
+    pub bytes: ffi::Slice<'a, u8>,
+}
+
+#[ffi]
+pub fn pattern_ffi_slice_in_struct(x: UseSliceByteInStruct) -> u32 {
+    x.bytes.as_slice().len() as u32
+}

